parsing/lexer.mll

```diff
@@ -124,15 +124,27 @@ let store_escaped_char lexbuf c =
 let store_escaped_uchar lexbuf u =
   if in_comment () then store_lexeme lexbuf else store_string_utf_8_uchar u
 
+let with_string f lexbuf =
```

Comment:
Could you use a clearer name for `f`? `consume_string`?

Reply:
Actually even `with_string` is a confusing name; usually we call `with_foo` functions of the form `init -> (foo -> 'a) -> 'a`. What about `build_string_token` or just `string_token`?

Reply:
It's called `with_string` because it's exactly the same as `with_comment` which is defined just underneath. Should I also change that function ?

Reply:
You have a semi-valid argument for leaving the code exactly as ugly as it currently is. I think you should do as you think is best.

Reply:
Renamed to `wrap_(string|comment)_lexer`.